### DISCUSSION TITLE
antd-mobile的InfiniteScroll 无限滚动碰到antd的table组件的scroll属性会触发无限滚动刷新，是否有办法规避？

### DISCUSSION THREAD

**Discussion by githubqh** [ant-design/ant-design-mobile#5398] [Q&A] [answered]
### Version of antd-mobile

all

### Browser and its version

all

### Sandbox to reproduce

https://codesandbox.io/s/ke-kong-de-shai-xuan-he-pai-xu-antd-4-21-3-forked-fhwldg?file=/demo.js

### What happened?

antd-mobile的InfiniteScroll 无限滚动碰到antd的table组件的scroll属性会触发无限滚动刷新，是否有办法规避？

### Relevant log output

![image](https://user-images.githubusercontent.com/105548468/175475202-d706363a-cade-4865-bbf3-47f948bb19dd.png)

**Comment by awmleer** [comment]
看起来是一个组件用法上的疑问，我把它转成 discussion 吧

**Comment by GOWxx** [comment]
```
用户想看到新的数据时，可以上滑页面自动加载数据。

当 hasMore 属性为 true 时，用户页面滚动到底部 threshold (默认为 250px)时无限滚动组件会调用定义的 loadMore 函数。
```

你给的 `scroll={{ y: 450 }}` , 是会导致 `InfiniteScroll` 始终符合触发条件。

**Reply by githubqh** [reply]
> ```
> 用户想看到新的数据时，可以上滑页面自动加载数据。
> 
> 当 hasMore 属性为 true 时，用户页面滚动到底部 threshold (默认为 250px)时无限滚动组件会调用定义的 loadMore 函数。
> ```
> 
> 你给的 `scroll={{ y: 450 }}` , 是会导致 `InfiniteScroll` 始终符合触发条件。

你好，不太明白threshold (默认为 250px)和scroll为什么会冲突，有什么办法可以规避吗

**Reply by GOWxx** [reply]
帮你改了下，可以看这个 [demo](https://codesandbox.io/s/ke-kong-de-shai-xuan-he-pai-xu-antd-4-21-3-forked-crg6nc?file=/demo.js)

你只需要简单的把 `<Table />` 和 `<InfiniteScroll />` 包裹在一个 `div`里，并给 `div` 设置合适的高度和超出有滚动条就可以了。
这个合适的高度由你的业务场景决定。例如你期望的 `450px`.

```ts
    <div
      className="App"
      style={{
        height: `450px`,
        overflowY: "auto"
      }}
    >
      <Table  />
      <InfiniteScroll threshold={0} />
    </div>
```

**Reply by githubqh** [reply]
> 帮你改了下，可以看这个 [demo](https://codesandbox.io/s/ke-kong-de-shai-xuan-he-pai-xu-antd-4-21-3-forked-crg6nc?file=/demo.js)
> 
> 你只需要简单的把 `<Table />` 和 `<InfiniteScroll />` 包裹在一个 `div`里，并给 `div` 设置合适的高度和超出有滚动条就可以了。 这个合适的高度由你的业务场景决定。例如你期望的 `450px`.
> 
> ```ts
>     <div
>       className="App"
>       style={{
>         height: `450px`,
>         overflowY: "auto"
>       }}
>     >
>       <Table  />
>       <InfiniteScroll threshold={0} />
>     </div>
> ```

非常感谢，但是如果去掉scroll属性，表格向下拉的时候就不能固定表头了，还有其他办法嘛？

**Comment by GOWxx** [comment, marked as answer]
> 非常感谢，但是如果去掉scroll属性，表格向下拉的时候就不能固定表头了，还有其他办法嘛？

又改了下，[可以看看这个 demo](https://codesandbox.io/s/ke-kong-de-shai-xuan-he-pai-xu-antd-4-21-3-forked-msbdgo?file=/demo.js)
思路是想办法把 `<InfiniteScroll />` 放入 `<Table />` 内的底部

```ts
      <Table
        scroll={{ y: 450 }}
        summary={() => (
          <div>
            <InfiniteScroll threshold={0} />
          </div>
        )}
      />
```

我目前看到 `summary` 比较合适
或者有任何其他能让它置于表格最后一行的办法应该都可以。

**Reply by githubqh** [reply]
对了！！非常感谢~大神
